Bug: reinhardt-test/msw fails to compile against web-sys 0.3.98 (Headers::new_with_str_mapping removed)

## Summary

`reinhardt-test/msw/interceptor.rs:183` calls `web_sys::Headers::new_with_str_mapping`, which does not exist in `web-sys` 0.3.98 (the version currently resolved in workspace dependencies). This makes the entire `msw` feature unusable from any downstream consumer that targets `wasm32-unknown-unknown` with current web-sys versions.

## Reproduction

In a downstream WASM consumer with `reinhardt-test = { version = "0.1.0-rc.28", features = ["msw"] }` dev-dependency:

```bash
wasm-pack test --headless --chrome  # or `cargo build --tests --target wasm32-unknown-unknown`
```

Errors:

```
error[E0271]: type mismatch resolving `<{closure} as FnOnce<...>>::Output == Result<Headers, JsValue>`
   --> .../reinhardt-test-0.1.0-rc.28/src/msw/interceptor.rs:183:46
    |
182 |     Headers::new_with_str_sequence_sequence(&h)
183 |         .or_else(|_| Headers::new_with_str_mapping(&h))
    |              -------------- ^^^^^^^^^^^^^^ expected `JsValue`, found `Object`

error[E0599]: no function or associated item named `new_with_str_mapping` found for struct `Headers` in the current scope
   --> .../reinhardt-test-0.1.0-rc.28/src/msw/interceptor.rs:183:31
note: if you're trying to build a new `Headers` consider using one of the following associated functions:
      Headers::new
      Headers::new_with_headers
      Headers::new_with_str_sequence_sequence
      Headers::new_with_record_from_str_to_str
```

## Affected Code

`crates/reinhardt-test/src/msw/interceptor.rs` lines 182–183:

```rust
Headers::new_with_str_sequence_sequence(&h)
    .or_else(|_| Headers::new_with_str_mapping(&h))
```

The fallback handles the `Record<string, string>` form (per Fetch spec) when the primary array form fails. `new_with_str_mapping` was the older web-sys binding for this; current web-sys exposes `Headers::new_with_record_from_str_to_str(init: &Object)` for the same purpose.

## Suggested Fix

Replace the deprecated/removed method with `new_with_record_from_str_to_str`, casting `&h` (currently `&JsValue`) to `&Object` via `JsCast::dyn_ref::<Object>()`. Roughly:

```rust
} else if let Some(obj) = h.dyn_ref::<Object>() {
    Headers::new_with_str_sequence_sequence(&h)
        .or_else(|_| Headers::new_with_record_from_str_to_str(obj))
        .ok()
        .map(|normalized| extract_headers(&normalized))
} else {
    None
}
```

(Exact form depends on `extract_headers`'s expected input; the maintainer is better positioned to choose the precise shape.)

## Impact

Blocks all downstream consumers of the typed MSW API (`MockServiceWorker`, `MockableServerFn`, etc.) on wasm32 with current web-sys versions. Discovered while migrating `examples-tutorial-basis` off the deprecated `mock_server_fn` API in #4275 / PR #4279.

## Environment

- Rust toolchain: 1.94.1-aarch64-apple-darwin
- Target: wasm32-unknown-unknown
- web-sys: 0.3.98 (resolved via workspace deps)
- wasm-pack: 0.13.x
- reinhardt-test: 0.1.0-rc.28 (both crates.io published and local workspace HEAD)

## Related

- Refs #4275, PR #4279 (the consumer that surfaced this)
- Refs #3283 (the original MSW feature issue, closed in rc.16)

🤖 Reported during local wasm-pack test verification of PR #4279.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -135,6 +135,10 @@ di = ["reinhardt-di", "reinhardt-di/params", "reinhardt-di/macros", "reinhardt-d
 test = ["reinhardt-test"]
 testcontainers = ["test", "reinhardt-test/testcontainers"]
 server-fn-test = ["test", "reinhardt-test/server-fn-test"]
+# MSW (Mock Service Worker) for WASM testing — exposes
+# `reinhardt::test::msw::MockServiceWorker` and triggers `MockableServerFn`
+# impl generation for `#[server_fn]` in downstream crates.
+msw = ["test", "pages", "reinhardt-test/msw", "reinhardt-pages/msw"]
 openapi = ["reinhardt-rest", "reinhardt-rest/openapi", "rest", "dep:inventory"]
 openapi-router = ["reinhardt-openapi", "openapi", "reinhardt-commands/openapi-router"]
 reinhardt-browsable-api = ["reinhardt-rest", "reinhardt-rest/browsable-api"]

--- a/crates/reinhardt-test/src/msw/interceptor.rs
+++ b/crates/reinhardt-test/src/msw/interceptor.rs
@@ -179,10 +179,21 @@ mod wasm_impl {
 							// Normalize plain objects and arrays by constructing a Headers
 							// instance, which accepts both `Record<string, string>` and
 							// `[string, string][]` forms per the Fetch spec.
-							Headers::new_with_str_sequence_sequence(&h)
-								.or_else(|_| Headers::new_with_str_mapping(&h))
-								.ok()
-								.map(|normalized| extract_headers(&normalized))
+							//
+							// (Fixes #4287) web-sys 0.3.98 removed
+							// `Headers::new_with_str_mapping`. Fall back through
+							// `new_with_record_from_str_to_str` when the input is a plain
+							// object (`js_sys::Object`); skip the fallback otherwise.
+							let primary = Headers::new_with_str_sequence_sequence(&h);
+							let normalized = match primary {
+								Ok(h) => Some(h),
+								Err(_) => h
+									.dyn_ref::<js_sys::Object>()
+									.and_then(|obj| {
+										Headers::new_with_record_from_str_to_str(obj).ok()
+									}),
+							};
+							normalized.map(|n| extract_headers(&n))
 						}
 					})
 					.unwrap_or_default()
@@ -253,9 +264,12 @@ mod wasm_impl {
 		let func: &js_sys::Function = original.unchecked_ref();
 		// Use the window (or globalThis) as the receiver to avoid "Illegal invocation"
 		// errors that occur when calling `window.fetch` with `this = null`.
+		// (Fixes #4287) `js_sys::global()` in newer js-sys returns `Object` rather
+		// than `JsValue`, so adapt it through `JsValue::from` before the
+		// `Option::unwrap_or_else` chain that expects a uniform `JsValue` output.
 		let receiver = web_sys::window()
 			.map(JsValue::from)
-			.unwrap_or_else(js_sys::global);
+			.unwrap_or_else(|| JsValue::from(js_sys::global()));
 		let result = if !init.is_undefined() && !init.is_null() {
 			func.call2(&receiver, input, init)?
 		} else {

